backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java

```diff
@@ -116,29 +114,20 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
 
         INodeStyle style = nodeDescription.getStyleProvider().apply(nodeVariableManager);
 
-        Position position = nodePositionProvider.getPosition(nodeId, optionalPreviousNode, this.props.getPreviousParentElement(), nodeSizeProvider, style, this.props.getParentAbsolutePosition());
+        var borderNodes = this.getBorderNodes(optionalPreviousNode, nodeVariableManager, nodeId);
+        var childNodes = this.getChildNodes(optionalPreviousNode, nodeVariableManager, nodeId);
 
-        Position absolutePosition = this.computeAbsolutePosition(position, this.props.getParentAbsolutePosition());
-        var borderNodes = this.getBorderNodes(optionalPreviousNode, nodeVariableManager, nodeId, absolutePosition);
-        var childNodes = this.getChildNodes(optionalPreviousNode, nodeVariableManager, nodeId, absolutePosition);
-
-        Size size;
-        if (childNodes.isEmpty()) {
-            size = nodeSizeProvider.getSize(style, childNodes);
-        } else {
-            size = optionalPreviousNode.map(Node::getSize).orElseGet(() -> nodeSizeProvider.getSize(style, childNodes));
-        }
-
-        NodeLabelPositionProvider labelBoundsProvider = new NodeLabelPositionProvider(type, size);
         LabelDescription labelDescription = nodeDescription.getLabelDescription();
         nodeVariableManager.put(LabelDescription.OWNER_ID, nodeId);
-        LabelComponentProps labelComponentProps = new LabelComponentProps(nodeVariableManager, labelDescription, optionalPreviousLabel, labelBoundsProvider, LabelType.INSIDE_CENTER.getValue());
+        LabelComponentProps labelComponentProps = new LabelComponentProps(nodeVariableManager, labelDescription, optionalPreviousLabel, LabelType.INSIDE_CENTER.getValue());
         Element labelElement = new Element(LabelComponent.class, labelComponentProps);
         List<Element> nodeChildren = new ArrayList<>();
         nodeChildren.add(labelElement);
         nodeChildren.addAll(borderNodes);
         nodeChildren.addAll(childNodes);
 
+        Position position = optionalPreviousNode.map(Node::getPosition).orElse(Position.UNDEFINED);
+        Size size = optionalPreviousNode.map(Node::getSize).orElse(Size.UNDEFINED);
```

Comment:
Are you sure that this will not add back the bug in which a semantic change cannot change the size of an element?

Reply:
Refactored in the last commit [285]